src/js/module/toolbox/sidebar.js

```diff
@@ -1,58 +1,97 @@
+import { Event } from 'const';
 
-const ToolboxViewSidebar = View.extend({
+const USER_ACTIVE_CLASS = 'is-user-active';
+const ACTIVITY_TIMEOUT_VALUE = 1500;
+
+export default View.extend({
   className: 'plantingjs-toolbox',
-  template: require('./sidebar.hbs'),
+  events: {
+    'dragstart .js-draggable-object': 'onDragStart',
+    'mouseenter': 'onMouseEnter',
+    'mouseleave': 'onMouseLeave',
+  },
+  userActivityTimeout: null,
+  mouseOver: false,
+
+  initialize() {
+    const objectsIds = range(this.manifesto().getCopy('toolboxobjects').length);
+    const objectsProjs = map(this.manifesto().getCopy('toolboxobjects'), ({ projections }) => projections);
+    const objectsData = chain(objectsIds)
+      .zip(objectsProjs)
+      .map(([ objectId, projections ]) => ({ objectId, projections, currentProjection: 0 }))
+      .value();
 
-  initialize: function initialize() {
-    const objectsIds = underscore.range(this.manifesto().getCopy('toolboxobjects').length);
-    const objectsProjs = underscore.map(this.manifesto().getCopy('toolboxobjects'), function(object) {
-      return object.projections;
+    this.collection = new Collection(objectsData, {
+      model: ToolboxModel,
+      app: this.app,
     });
-    const objectsData = underscore.zip(objectsIds, objectsProjs);
-
-    this.collection = new ToolboxCollection(underscore.map(objectsData, function(objectData) {
-      return {
-        objectId: objectData[0],
-        projections: objectData[1],
-        currentProjection: 0,
-      };
-    }), {app: this.app});
-
-    this.objects = this.collection.map(function(toolboxObjectModel) {
-      return new ToolboxViewObject({
-        model: toolboxObjectModel,
-        app: this.app,
-      });
-    }, this);
-    this.app.on(Const.Event.START_PLANTING, function() {
-      this.$el.show();
-    }, this);
-    this.render( this.objects );
-  },
-
-  render: function render(objects) {
-    const $template = jquery('<div />').append(this.template());
-    const $list = $template.find('.plantingjs-toolboxobject-container');
-
-    objects.forEach(function(object) {
-      $list.append(object.$el);
+    this.render();
+    this.app.on(Event.START_PLANTING, () => {
+      this.renewUserActivity(ACTIVITY_TIMEOUT_VALUE * 2);
     });
-    this.$el.append($template);
   },
 
-  hide: function hide() {
-    this.$el.hide();
+  render() {
+    let objects = this.collection.map((model) => {
+      const {
+        projections: [image],
+        objectId,
+      } = model.attributes;
+      const cid = model.cid;
+
+      return { image, objectId, cid };
+    });
+
+    objects = objects.concat(objects, objects, objects);
+    this.$el.html(template({ objects }));
+    this.makeObjectsDraggable();
   },
 
-  show() {
-    this.$el.show();
+  makeObjectsDraggable() {
+    const config = {
+      containment: '.plantingjs-overlay',
+      helper: 'clone',
+      appendTo: '.plantingjs-overlay',
+      zIndex: 10000,
```

Comment:
Does `zIndex` should be set to such big number. AFAIR in previous PR button has `zIndex` set to `101`. Do we really want for dropped object to be rendered above this button?